File: HoTT-Model/src/main/java/de/treichels/hott/util/ModelLoader.java
package de.treichels.hott.util;

import de.treichels.hott.model.BaseModel;

import java.io.IOException;

public interface ModelLoader {
    BaseModel getModel() throws IOException;

    byte[] getModelData() throws IOException;

    void onCancel();

    void onOpen();

    void onReload();
}

File: HoTT-Model/src/main/java/de/treichels/hott/util/ModelLoader.kt
package de.treichels.hott.util

import de.treichels.hott.model.BaseModel
import java.io.IOException

interface ModelLoader {

    val model: BaseModel?
    @Throws(IOException::class) get


    val modelData: ByteArray?
    @Throws(IOException::class) get

    fun onCancel()
    fun onOpen()
    fun onReload()
}